ngm/__init__.py

```diff
@@ -95,11 +95,22 @@ def dominant_eigen(X: np.ndarray, norm: str = "L1") -> DominantEigen:
     value = e.eigenvalues[i]
     vector = _ensure_positive_array(e.eigenvectors[:, i])
 
+    # check for negative values
     if not value > 0:
         raise RuntimeError(f"Negative dominant eigenvalue: {value}")
     if not all(vector >= 0):
         raise RuntimeError(f"Negative dominant eigenvector values: {vector}")
 
+    # check for complex numbers
+    if np.iscomplex(value) and value.imag != 0:
+        raise RuntimeError("Complex numbers in eigenvalue")
+    if np.iscomplex(vector).any() and not np.all(vector.imag == 0):
+        raise RuntimeError("Complex numbers in eigenvector")
+
+    # Ensure the value and vector are real
```

Comment:
```suggestion
    # Ensure the value and vector are dtype float (not complex)
```
trying to draw the distinction between real-valued versus real-typed